packages/provider-utils/src/validator.ts

```diff
@@ -19,7 +20,7 @@ export type Validator<OBJECT = unknown> = {
    * Optional. Validates that the structure of a value matches this schema,
    * and returns a typed version of the value if it does.
    */
-  readonly validate?: (value: unknown) => ValidationResult<OBJECT>;
+  readonly validate?: (value: unknown) => ValidationResult<OBJECT> | PromiseLike<ValidationResult<OBJECT>>;
```

Comment:
no sure if making the `validate()` method optionally async could cause any problems down the road, but it made the transition easier.